File: examen-backend/src/main/java/com/rei/examenbackend/service/TodoService.java
package com.rei.examenbackend.service;

import com.rei.examenbackend.dto.todo.TodoRequest;
import com.rei.examenbackend.dto.todo.TodoResponse;
import com.rei.examenbackend.model.User;

import java.util.List;

public interface TodoService {
    List<TodoResponse> list(User user);
    TodoResponse create(TodoRequest request, User user);
    TodoResponse update(Long id, TodoRequest request, User user);
    TodoResponse toggle(Long id, User user);
    void delete(Long id, User user);
}

File: examen-frontend/src/app/shared/features/todos/services/todo.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiConfigService } from '../../../../core/services/api-config.service';

export interface Todo {
  id: number;
  title: string;
  completed: boolean;
  dueAt: string;
}

@Injectable({
  providedIn: 'root'
})
export class TodoService {
  private baseUrl: string;

  constructor(
    private http: HttpClient,
    private api: ApiConfigService
  ) {
    this.baseUrl = this.api.endpoint('todos');
  }

  getAll(): Observable<Todo[]> {
    return this.http.get<Todo[]>(this.baseUrl);
  }

  create(todo: Partial<Todo>): Observable<Todo> {
    return this.http.post<Todo>(this.baseUrl, todo);
  }

  update(id: number, todo: Partial<Todo>): Observable<Todo> {
    return this.http.patch<Todo>(`${this.baseUrl}/${id}`, todo);
  }

  toggle(id: number): Observable<Todo> {
    return this.http.patch<Todo>(`${this.baseUrl}/${id}/toggle`, {});
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
